src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotSerializeTypeWithPointerFieldsTests.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class DoNotSerializeTypeWithPointerFieldsTests : DiagnosticAnalyzerTestBase
```

Comment:
Would be good to see unit tests for:
- Properties
- Arrays of pointers
- Collections of pointers
- Generic types

Basically all the edge cases tested in DoNotReferSelfInSerializableClassTests.cs.